<?xml version="1.0"?>
<launch>

<!-- Start rosbag with launchfile-->
<arg name="run_bag" default="false" />
<!-- Absolute Path to map to be used for localization. The map needs to be built by graph_fuser package -->
<arg name="map_file_path" default="/home/daniel/.ros/maps/offmichigan_gt=0_submap=1_sizexy=150_Z=15_intrchR=20_compR=10_res=0.8_maxSensd=130_keyF=1_d=1_deg=150_alpha=0_dl=0_xyzir=0_mpsu=0_mnpfg=6kmnp0.map" />
<arg name="rviz_enabled" default="true" />

<!-- Include visualization in rviz -->
<group if="$(arg rviz_enabled)">
<include file="$(find graph_map)/launch/visualize_graph_fuser.launch" >
<arg name="localization" value="true" />
<arg name="michigan" value="true" />
</include>
</group>
<!-- Run monte carlo localization -->
<node name="dl_localization" pkg="self_localization" type="dl_localization" output="screen">

<!-- select one of the following types-->

<param name="gt_tf" value="/state_base_link" /> <!--robot_odom_link-->
<param name="base_tf" value="/state_base_link" />

<param name="dataset" value="michigan" />
<param name="localisation2D" value="false" />
<param name="enable_localisation" value="true" />
<param name="gt_initialize" value="true" />
<param name="map_localizaiton" value="true" />



<!-- Topic of laser scanner -->
<param name="points_topic" value="/velodyne_points" />

<!-- pass path to graph_map (.map)-->
<param name="map_file" value="$(arg map_file_path)"  />
<param name="sensor_pose_x" value="0.002" />
<param name="sensor_pose_y" value="-0.004" />
<param name="sensor_pose_z" value="-0.957" />
<param name="sensor_pose_r" value="0.01408480706" />
<param name="sensor_pose_p" value="0.00289724655" />
<param name="sensor_pose_t" value="-1.58306599144" />

<param name="init_ex" value="0" /> <!--3.1415-->

<!-- Choose weather to initiate pose to pose_init_<x,y,t> or the data of /<gt_topic> -->

<param name="resolution_local_factor" value="1.3" />
<param name="particle_count" value="500" />
<param name="show_pose" value="true"/>
<param name="fraction" value="1.0"/>
<param name="force_SIR" value="true" />
<param name="z_filter_height" value="-1000" />

</node>


</launch>
